### ISSUE TITLE
Each Namespace Holds a Connection

### ISSUE THREAD

**Issue by dave-apex** [socketio/socket.io-postgres-adapter#3]
First, we love this package and the opportunities that it opens up for us. Great job :).

We have an app that started having database connection problems after adding socket.io. After doing some digging it appears that the issue is that the postgres adapter holds a connection for each namespace. That would be fine for many apps but our design is based on custom namespaces. It appears that every namespace that is accessed by some user will then hold a connection indefinitely. Is that the expected behavior? Are there any patterns that could free up these connections when the last user disconnects from the namespace?

**Comment by darrachequesne** [comment]
> It appears that every namespace that is accessed by some user will then hold a connection indefinitely.

Yes, that's indeed the current behavior, but I think we could change it.

As a temporary workaround, you can close the connection when the last user disconnects:

```js
const namespace = io.of("/the-namespace");

namespace.on("connection", (socket) => {
  socket.on("disconnect", () => {
    const isLast = namespace.sockets.size === 0;
    if (isLast) {
      namespace.adapter.close();
    }
  });
});
```

**Comment by dave-apex** [comment]
Thanks makes a lot of sense. In the meantime, we have moved from Namespaces to Rooms and that is working fine for us.

**Comment by darrachequesne** [comment]
This should be fixed in version 0.3.0, implemented in https://github.com/socketio/socket.io-postgres-adapter/commit/651e28169185d91c7a1a86152d21aa265d5500f2.
